pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -850,7 +851,7 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) []*model.Servi
 		pod := c.pods.getPodByIP(proxyIP)
 		if workload, f := c.workloadInstancesByIP[proxyIP]; f {
 			return c.hydrateWorkloadInstance(workload)
-		} else if pod != nil {
+		} else if pod != nil && proxy.Metadata.Labels[constants.TestVMLabel] == "" {
```

Comment:
It does not feel right to have a check for the test label in our non-test code. This seems to indicate either a bug in the test or a bug in our real code that is being masked? Or is this because we do the fake pod thing?

Reply:
It's because of the fake pod thing. I'm not sure how else to make them bypass this block unless we have some node metadata to indicate "this proxy is on a vm"

Reply:
This looks much better, and please add a comment

Reply:
Can we abstract this to a proxy function like` proxy.isVirtualMachine()` and have this logic there?